### PR TITLE
api: Add Id in the ExternalStorageClusterStatus struct

### PR DESCRIPTION
**Pull request by iamniting**
Id field is required to keep/save the UID of a storage consumer CR from
the provider cluster in the status. It will be used/passed while making
the API calls (offboarding, upscaling/downscaling) from the consumer
cluster after onboarding is successful.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

cc @nb-ohad @sp98 @kesavanvt

### CODE REVIEW

**Comment by sp98** [comment]
Please update the description and commit ID to convey why is it required to keep/save the UID

File: api/v1/storagecluster_types.go [hunk 1]
```diff
@@ -177,6 +177,9 @@ type ExternalStorageClusterStatus struct {
 	// GrantedCapacity Will report the actual capacity
 	// granted to the consumer cluster by the provider cluster.
 	GrantedCapacity resource.Quantity `json:"grantedCapacity,omitempty"`
+
+	// Id will describe the Id of an connected ExternalStorageCluster
+	Id string `json:"id,omitempty"`
```
Comment (sp98) [inline api/v1/storagecluster_types.go:182]: `Id` sounds a bit vague. It does not convey which ID is being referred.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: api/v1/storagecluster_types.go (hunk 1 above)
Comment (sp98) [reply api/v1/storagecluster_types.go:182]: Suggestions:   `ProviderID` ,`OCSProviderID`,  `StorageClusterID`

**Comment by iamniting** [comment]
> @iamniting Please update the description and commit message to convey why is it required to keep/save the UID

updated

File: api/v1/storagecluster_types.go (hunk 1 above)
Comment (iamniting) [reply api/v1/storagecluster_types.go]: What do you think it should be? Can you list down the names you have in your mind?

Comment (kesavanvt) [reply api/v1/storagecluster_types.go]: It should be `ConsumerId` as its the identifier for the consumer

Comment (sp98) [reply api/v1/storagecluster_types.go]: yeah. Ignore my above comment.  I thought its the storageCluster ID.  ConsumerID looks good.

Comment (iamniting) [reply api/v1/storagecluster_types.go]: done

File: config/crd/bases/ocs.openshift.io_storageclusters.yaml [hunk 2]
```diff
@@ -4772,6 +4772,9 @@ spec:
                       to the consumer cluster by the provider cluster.
                     pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                     x-kubernetes-int-or-string: true
+                  id:
+                    description: ConsumerId will describe the Id of an connected ExternalStorageCluster
```
Comment (nb-ohad) [inline config/crd/bases/ocs.openshift.io_storageclusters.yaml:4776]:
```suggestion
                    description: ConsumerId will hold the identity of this cluster inside the attached provider cluster
```

File: deploy/bundle/manifests/storagecluster.crd.yaml [hunk 3]
```diff
@@ -4770,6 +4770,9 @@ spec:
                       to the consumer cluster by the provider cluster.
                     pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                     x-kubernetes-int-or-string: true
+                  id:
+                    description: ConsumerId will describe the Id of an connected ExternalStorageCluster
```
Comment (nb-ohad) [inline deploy/bundle/manifests/storagecluster.crd.yaml:4774]:
```suggestion
                    description: ConsumerId will hold the identity of this cluster inside the attached provider cluster
```

File: deploy/csv-templates/crds/ocs/ocs.openshift.io_storageclusters.yaml [hunk 4]
```diff
@@ -4772,6 +4772,9 @@ spec:
                       to the consumer cluster by the provider cluster.
                     pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                     x-kubernetes-int-or-string: true
+                  id:
+                    description: ConsumerId will describe the Id of an connected ExternalStorageCluster
```
Comment (nb-ohad) [inline deploy/csv-templates/crds/ocs/ocs.openshift.io_storageclusters.yaml:4776]:
```suggestion
                    description: ConsumerId will hold the identity of this cluster inside the attached provider cluster
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Review by sp98 — APPROVED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**